### ISSUE TITLE
Is Spring cloud Pub/Sub library automatically reconnects to GCP Pub/Sub API, If GCP Pub/Sub service goes down come back again?

### ISSUE THREAD

**Issue by veereshwaran** [spring-cloud/spring-cloud-gcp#2031]
I am using Spring cloud Pub/Sub library in our micro-services

Our micro-services listening respective Pub/Sub subscriptions. 

After GCP service outage on 11th Nov, our any micro-service not listening topic and message stayed subscriptions and not received in our micro-service.

After restart only, our micro-services start receiving messages from GCP Pub/Sub subscriptions.

I am using default spring-cloud binding configuration and 
```
spring.cloud:
  gcp:
    project-id: ${GCP_PROJECT_ID}
    credentials.location: file:${GCP_CREDENTIALS_LOCATION}
  stream:
    default.contentType: application/json
    bindings:
      my-topic:
        destination: ${TOPIC_NAME}
```

Is there any options to auto-reconnect Pub/Sub this kind of service outage?

**Comment by meltsufin** [comment]
@veereshwaran It's a good question. Since Spring Cloud GCP simply delegates to the Pub/Sub client library which handles the connection details, it's more of a question for them. I posted it there:
https://github.com/googleapis/google-cloud-java/issues/6763

**Comment by elefeint** [comment]
@veereshwaran I believe this issue was resolved in #1807.

You can use `spring.cloud.gcp.pubsub.keepAliveIntervalMinutes` property to control the frequency of the keepalive ping.

**Comment by veereshwaran** [comment]
Hi @elefeint 

Previously I used spring-cloud-gcp-starter-pubsub version 1.1.0.RELEASE

Now I upgraded pubsub version to 1.2.1.RELEASE

In the latest version, It will wait 5 mins to attempt reconnect after connection broken. 

Now I got a clear answer why it take 5 mins to re-establish connection, after seeing the PR #1807.

For now 5 mins interval time enough. 

Thanks for your update
